### PR TITLE
Bloom Options using Database

### PR DESCRIPTION
**Pull request by cdsupina**
- Added options table to db save file
- Added GameOptions resource
  - Defaults to options for best performance
  - With storage feature, loads options from  db
- Used bloom_intensity game option to control bloom throughout the game

### CODE REVIEW

File: crates/thetawave_storage/src/core.rs [hunk 1]
```diff
@@ -56,9 +57,16 @@ pub(super) fn setup_db(conn: Connection) -> rusqlite::Result<()> {
         PRIMARY KEY (userId, enemyMobType)
     )"
     );
+
+    let create_options_table_sql = format!(
+        "CREATE TABLE IF NOT EXISTS {OPTIONS_TABLE_NAME} (
```
Comment (cdsupina) [inline crates/thetawave_storage/src/core.rs:63]: Create primary key for the row.

File: crates/thetawave_storage/src/core.rs [hunk 2]
```diff
@@ -56,9 +57,27 @@ pub(super) fn setup_db(conn: Connection) -> rusqlite::Result<()> {
...
     conn.execute(&create_enemies_killed_table_sql, [])
         .map(|_| ())?;
+    conn.execute(&create_options_table_sql, []).map(|_| ())?;
+
+    // insert a default options row if it is not in the db
+    let default_options_row_exists: Option<u32> = conn
+        .query_row("SELECT 1 FROM Options LIMIT 1", [], |row| row.get(0))
+        .optional()?;
+    if default_options_row_exists.is_none() {
+        conn.execute("INSERT INTO Options DEFAULT VALUES", [])?;
```
Comment (varoonp123) [inline crates/thetawave_storage/src/core.rs]: To just make sure a row is always in the table:
```rust
// Ensure that the default options profile is _always_ usable. 
let upsert_default_gameopts_sql = format!("INSERT OR REPLACE INTO {} (optionsProfileId, bloomEnabled, bloomIntensity) VALUES (?1, TRUE, 1.0) ON CONFLICT REPLACE", OPTIONS_TABLE_NAME);
conn.execute(upsert_default_gameopts_sql. [DEFAULT_OPTIONS_PROFILE_ID])
```

File: src/options/mod.rs [hunk 3]
```diff
@@ -51,6 +52,41 @@ impl GameInitCLIOptions {
...
+        (With<Camera2d>, Without<Camera3d>),
+    >,
+    mut camera_3d_query: Query<
+        (&mut Camera, &mut Tonemapping),
+        (With<Camera3d>, Without<Camera2d>),
+    >,
+) {
+    let (mut camera_2d, mut tonemapping_2d) = match camera_2d_query.get_single_mut() {
+        Ok(camera) => camera,
+        Err(_) => panic!("Didn't find exactly one 2D camera to apply bloom settings to."),
```
Comment (varoonp123) [inline src/options/mod.rs]: What.


Why.


How.



Maybe 
```rust
if let ((mut camera_2d, mut tonemapping_2d), (mut camera_3d, tonemapping_3d)) = (camera_2d_query.get_single_mut(), camera_3d_query.get_single_mut()) {

}
else {
    error!("Failed to get singleton 2d and 3d cameras to apply game opts");
}
```

File: crates/thetawave_storage/src/core.rs (hunk 2 above)
Comment (cdsupina) [reply crates/thetawave_storage/src/core.rs]: It looks like having "INSERT OR REPLACE" and "ON CONFLICT REPLACE" doesn't work. Looks like having just "INSERT OR REPLACE" is the way to go, let me know if I'm missing something.

**Review by varoonp123 — APPROVED** [review]
i thought i already approved this

**Pull request merged**